casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -151,11 +152,11 @@ class RuntimeManager private (val emptyStateHash: ByteString, runtimeContainer:
       terms match {
         case deploy +: rem =>
           runtime.space.reset(hash)
-          val availablePhlos             = CostAccount(Integer.MAX_VALUE)
-          implicit val costAccountingAlg = CostAccountingAlg.unsafe[Task](availablePhlos) //FIXME this needs to come from the deploy params
-          val (phlosLeft, errors)        = injAttempt(deploy, runtime.reducer, runtime.errorLog)
-          val cost                       = phlosLeft.copy(cost = availablePhlos.cost.value - phlosLeft.cost)
-          val newCheckpoint              = runtime.space.createCheckpoint()
+          val availablePhlos = Cost(Integer.MAX_VALUE) // FIXME: This needs to come from the deploy params
+          runtime.reducer.setAvailablePhlos(availablePhlos).runSyncUnsafe(1.second)
```

Comment:
This is very unfortunate, and I really can't get over it.

I see that the `set` is needed, because we only want to create the ChargingReducer and the undrlying CostAccountingAlg once.

But please remind me, what's the reason for us not being able to thread the CostAccountingAlg through implicit method parameters, and create it here?

I appreciate the fact that we can't initialize the CostAccountingAlg until we have the phloLimit. But then, **this is the exact place it should be created at**.

I can see now yet another reason why @rabbitonweb insists on passing capabilities via methods' type parameters instead of fixing them once in trait/class header.